lib/archethic_web/controllers/faucet_controller.ex

```diff
@@ -73,30 +73,34 @@ defmodule ArchEthicWeb.FaucetController do
          {:ok, last_index} <- ArchEthic.get_transaction_chain_length(last_address) do
       create_transaction(last_index, curve, recipient_address)
     else
-      {:error, _} = e ->
+      e ->
         e
     end
   end
 
   defp create_transaction(transaction_index, curve, recipient_address) do
-    Transaction.new(
-      :transfer,
-      %TransactionData{
-        ledger: %Ledger{
-          uco: %UCOLedger{
-            transfers: [
-              %UCOLedger.Transfer{
-                to: recipient_address,
-                amount: 10_000_000_000
-              }
-            ]
+    tx =
+      Transaction.new(
+        :transfer,
+        %TransactionData{
+          ledger: %Ledger{
+            uco: %UCOLedger{
+              transfers: [
+                %UCOLedger.Transfer{
+                  to: recipient_address,
+                  amount: 10_000_000_000
+                }
+              ]
+            }
           }
-        }
-      },
-      @pool_seed,
-      transaction_index,
-      curve
-    )
-    |> ArchEthic.send_new_transaction()
+        },
+        @pool_seed,
+        transaction_index,
+        curve
+      )
+
+    {Base.encode16(tx.address),
```

Comment:
You should encode before to render the view, here it's more a business function, so keep in binary. 
Also you should return `{:ok, tx.address}` only when the transaction is submitted